pkg/conversion/generator.go

```diff
@@ -458,12 +481,32 @@ func (g *generator) writeConversionForPtr(w io.Writer, inField, outField reflect
 		}
 	}
 
-	assignFormat := "if err := s.Convert(&in.%s, &out.%s, 0); err != nil {\n"
-	assignStmt := fmt.Sprintf(assignFormat, inField.Name, outField.Name)
-	if err := writeLine(w, indent, assignStmt); err != nil {
+	ifFormat := "if in.%s != nil {\n"
```

Comment:
I'm a little nervous that the else of this doesn't set out.%s = nil; out could be a pre-existing object...